server/src/main/java/org/elasticsearch/action/termvectors/TermVectorsFilter.java

```diff
@@ -218,8 +216,8 @@ public void selectBestTerms() throws IOException {
                 }
 
                 // now call on docFreq
-                long docFreq = getTermStatistics(topLevelTermsEnum, term).docFreq();
-                if (!isAccepted(docFreq)) {
+                long docFreq = foundTerm ? getTermStatistics(topLevelTermsEnum, term).docFreq() : 1;
```

Comment:
I'm curious to understand why you choose a small dummy frequency here instead of ignoring the term by using foundTerm in a condition before this and continue if its false. I probably don't understand this well enough, maybe the 1 freq is whats needed here but I'd like to understand better. Could you explain?

Reply:
I choose 1 to ensure that we keep the term if there are not enough terms present in the index that can be used for the query. We could also completely filter these terms but then they would never be returned in an artificial doc that retain all terms (and in such case I think it's better to retain them with a frequency of 1 since they appear in the provided doc) ?

Reply:
Thanks, that makes sense.